data/preprocess/util.py

```diff
@@ -252,10 +256,18 @@ def store_map(symbol_map: Dict, filename: str):
         for symbol, index in symbol_map.items():
             f.write(f"{index}\t{symbol}\n")
 
+def dump_buffer_to_file(buffer, fptr):
+    """Dump buffer to file."""
+    fptr.write(f"{buffer}")
 
-def write_triple(f, ent, rel, t, S, P, O):
+def write_triple(f, buffer, ent, rel, t, S, P, O):
     """Write a triple to a file. """
-    f.write(str(ent[t[S]]) + "\t" + str(rel[t[P]]) + "\t" + str(ent[t[O]]) + "\n")
+    buffer += f"{str(ent[t[S]])}\t{str(rel[t[P]])}\t{str(ent[t[O]])}\n"
```

Comment:
I think it is faster, to make the buffer a list of strings and finally join this list to a single string
```python
buffer_list = []
buffer_list.append(new_triple_string)

buffer = ""
buffer.join(buffer_list)
```